php/get-access-token

```diff
@@ -1,64 +1,106 @@
 #!/usr/bin/env php
 <?php
 require 'vendor/autoload.php';
-use GuzzleHttp\Client;
-use GuzzleHttp\HandlerStack;
-use GuzzleHttp\Subscriber\Oauth\Oauth1;
+use League\Oauth2\Client\Provider\GenericProvider;
 
-const OAUTH_URL = 'https://auth.aweber.com/1.0/oauth/';
 
-echo 'Enter your consumer key: ';
-$consumerKey = rtrim(fgets(STDIN), PHP_EOL);
-echo 'Enter your consumer secret: ';
-$consumerSecret = rtrim(fgets(STDIN), PHP_EOL);
+const OAUTH_URL = 'https://auth.aweber.com/oauth2/';
+const TOKEN_URL = 'https://auth.aweber.com/oauth2/token';
 
-// Create a Guzzle client configured to use OAuth for authentication
-$stack = HandlerStack::create();
-$client = new Client([
-    'base_uri' => OAUTH_URL,
-    'handler' => $stack,
-    'auth' => 'oauth'
-]);
 
-// Use your consumer key and secret to get a request token and secret.
-// Use "oob" as the oauth_callback to display the verifier so it can be copied.
-// See the Getting Started guide to redirect to a callback URL: https://api.aweber.com/#tag/Getting-Started
-$requestMiddleware = new Oauth1([
-    'consumer_key' => $consumerKey,
-    'consumer_secret' => $consumerSecret,
-    'token' => null,
-    'token_secret' => null,
-]);
-$stack->push($requestMiddleware);
-$res = $client->post('request_token', ['form_params' => ['oauth_callback' => 'oob']]);
-$params = [];
-parse_str($res->getBody(), $params);
+echo 'Do you wish to create(c) tokens or refresh(r) tokens? ';
+$createRefresh = rtrim(fgets(STDIN), PHP_EOL);
 
-// Log in to receive a verifier code for the request token
-echo "Go to this url: " . OAUTH_URL . "authorize?oauth_token={$params['oauth_token']}\n";
-echo 'Log in and paste the returned verifier code here: ';
-$verifier = rtrim(fgets(STDIN), PHP_EOL);
+if (strtoupper($createRefresh) == 'C') {
+    echo 'Enter your client id: ';
+    $clientId = rtrim(fgets(STDIN), PHP_EOL);
+    echo 'Enter your client secret: ';
+    $clientSecret = rtrim(fgets(STDIN), PHP_EOL);
 
-// Trade the request token and secret and verifier for an access code and secret
-$accessMiddleware = new Oauth1([
-    'consumer_key'    => $consumerKey,
-    'consumer_secret' => $consumerSecret,
-    'token' => $params['oauth_token'],
-    'token_secret' => $params['oauth_token_secret'],
-]);
-$stack->remove($requestMiddleware);
-$stack->push($accessMiddleware);
-$res = $client->post('access_token', ['form_params' => ['oauth_verifier' => $verifier]]);
-$credentials = [];
-parse_str($res->getBody(), $credentials);
+    $redirectUri = 'https://localhost';
+
+    $scopes = array(
+        'account.read',
```

Comment:
im wondering if we should make this configurable?

Reply:
Team decision was to just leave it hardcoded.